src/catalog/src/expr_cache.rs

```diff
@@ -205,6 +221,18 @@ impl ExpressionCache {
             .collect();
         self.durable_cache.try_set_many(&keys_to_remove).await?;
 
+        if compact_shard {
+            // The expression cache re-uses the catalog compaction dyncfgs because they are tightly
+            // related.
+            let fuel = CATALOG_FORCE_COMPACTION_FUEL.handle(dyncfgs);
+            let wait = CATALOG_FORCE_COMPACTION_WAIT.handle(dyncfgs);
+            let _handle = self.durable_cache.dangerous_compact_shard(
```

Comment:
I tried blocking on this, but the test would hang for a very long time. Probably because I'm reusing the catalog's wait configs.